modules/stvBidAdapter.js

```diff
@@ -0,0 +1,308 @@
+        Object.keys(videoParams)
+          .filter(key => includes(Object.keys(VIDEO_ORTB_PARAMS), key) && params[VIDEO_ORTB_PARAMS[key]] === undefined)
+          .forEach(key => payload.pfilter[VIDEO_ORTB_PARAMS[key]] = videoParams[key]);
+      }
+      if (Object.keys(payload.pfilter).length == 0) { delete payload.pfilter }
+
+      if (bidderRequest && bidderRequest.gdprConsent) {
+        payload.gdpr_consent = bidderRequest.gdprConsent.consentString;
+        payload.gdpr = bidderRequest.gdprConsent.gdprApplies;
+      }
+
+      if (params.bcat !== undefined) {
+        payload.bcat = deepAccess(bidderRequest.ortb2Imp, 'bcat') || params.bcat;
+      }
+      if (params.dvt !== undefined) {
+        payload.dvt = params.dvt;
+      }
+      if (isDev) {
+        payload.prebidDevMode = 1;
+      }
+
+      if (pbcode) {
+        payload.pbcode = pbcode;
+      }
+
+      payload.media_types = convertMediaInfoForRequest(mediaTypesInfo);
+
+      return {
+        method: 'GET',
+        url: endpoint,
+        data: objectToQueryString(payload),
+      };
+    });
+  },
+  interpretResponse: function(serverResponse, bidRequest) {
+    const bidResponses = [];
+    const response = serverResponse.body;
+    const crid = response.crid || 0;
+    const cpm = response.cpm / 1000000 || 0;
+    if (cpm !== 0 && crid !== 0) {
+      const dealId = response.dealid || '';
+      const currency = response.currency || 'EUR';
+      const netRevenue = (response.netRevenue === undefined) ? true : response.netRevenue;
+      const bidResponse = {
+        requestId: response.bid_id,
+        cpm: cpm,
+        width: response.width,
+        height: response.height,
+        creativeId: crid,
+        dealId: dealId,
+        currency: currency,
+        netRevenue: netRevenue,
+        ttl: config.getConfig('_bidderTimeout'),
+        meta: {
+          advertiserDomains: response.adomain || []
+        }
+      };
+      if (response.vastXml) {
+        bidResponse.vastXml = response.vastXml;
+        bidResponse.mediaType = 'video';
+      } else {
+        bidResponse.ad = response.adTag;
+      }
+
+      bidResponses.push(bidResponse);
+    }
+    return bidResponses;
+  },
+  getUserSyncs: function(syncOptions, serverResponses, gdprConsent, uspConsent) {
+    if (!serverResponses || serverResponses.length === 0) {
+      return [];
+    }
+
+    const syncs = []
+
+    let gdprParams = '';
+    if (gdprConsent) {
+      if ('gdprApplies' in gdprConsent && typeof gdprConsent.gdprApplies === 'boolean') {
+        gdprParams = `gdpr=${Number(gdprConsent.gdprApplies)}&gdpr_consent=${gdprConsent.consentString}`;
+      } else {
+        gdprParams = `gdpr_consent=${gdprConsent.consentString}`;
+      }
+    }
+
+    if (serverResponses.length > 0 && serverResponses[0].body.userSync) {
+      if (syncOptions.iframeEnabled) {
+        serverResponses[0].body.userSync.iframeUrl.forEach((url) => syncs.push({
```

Comment:
please add checks for existence of each property in the serverResponse array entry before running each iframe and pixel enabled sync loop